runtime/src/main/scala/akka/grpc/GrpcWeb.scala

```diff
@@ -0,0 +1,162 @@
+package akka.grpc
+
+import java.util.Base64
+
+import akka.grpc.Grpc.GrpcFramingDecoderStage
+import akka.grpc.GrpcProtocol._
+import akka.grpc.GrpcWebGen.{ marshaller, unmarshaller }
+import akka.grpc.scaladsl.headers
+import akka.http.javadsl.{ model => jmodel }
+import akka.http.scaladsl.model.HttpEntity.{ Chunk, ChunkStreamPart }
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{ `Accept-Encoding`, `Content-Encoding`, `Content-Type`, Accept }
+import akka.stream.scaladsl.Flow
+import akka.util.ByteString
+import ch.megard.akka.http.cors.scaladsl.model.HttpHeaderRange
+import ch.megard.akka.http.cors.scaladsl.settings.CorsSettings
+import io.grpc.{ Status, StatusException }
+
+import scala.collection.immutable
+
+trait GrpcWebVariant extends GrpcVariant {
```

Comment:
I wonder if the specific variant(/protocol)s should be in `akka.grpc.internal` - are they ever part of a user-facing API?

Reply:
I had toyed with them being optionally enabled in the service handler builder, but in the end kept is simple. Only the default CORS settings need a home in the user-facing API, and they could live in the service handler.

Reply:
Have moved these into .internal - there's no reason for them to be in .grpc at present I think.